segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/health/ZKHealthContributor.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.pravega.segmentstore.server.host.health;
+
+import io.pravega.shared.health.Health;
+import io.pravega.shared.health.Status;
+import io.pravega.shared.health.impl.AbstractHealthContributor;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.imps.CuratorFrameworkState;
+
+public class ZKHealthContributor  extends AbstractHealthContributor {
+    private CuratorFramework zk;
+
+    public ZKHealthContributor(CuratorFramework zk) {
```

Comment:
Add `@NotNull` to the argument - that will enforce that you aren't given a `null` value.

Tip: there's `@Nonnull`, `@Notnull` and `@NotNull` (at least in my classpath). Make sure you use the one from the `lombok` package. If in doubt, search for it elsewhere in the codebase and see which one we use.

Reply:
@dellThejas I asked that you add `@NonNull` to the argument to the constructor, not to the class-level variable.

What you did is OK if you also use `@RequiredArgsConstructor` on the class which will auto-generate the constructor for you. But if you have a method/constructor, you need to set that annotation on the argument to the method itself.

Reply:
^^ @dellThejas